# Modeling Modlule

The modeling module provides a single Python API for data scientists to take advantage of the capabilities of ArcGIS as part of demographic geographic data science workflows.

This first cell is largely a bunch of bubblegum and duct tape tying all the things together so I can use this notebook for prototyping and testing.

In [1]:
import importlib
import os
from pathlib import Path
import re
import sys

from dotenv import load_dotenv, find_dotenv
import pandas as pd

# load the "autoreload" extension so that code can change, & always reload modules so that as you change code in src, it gets loaded
%load_ext autoreload
%autoreload 2

# load environment variables from .env
load_dotenv(find_dotenv())

dir_src = Path.cwd().parent.parent/'src'

sys.path.insert(0, str(dir_src))

In [2]:
from arcgis.gis import GIS
from modeling import get_countries, Country

In [4]:
usa = Country('USA')

usa

<modeling.Country - USA (local 2020)>

Next, is is extremely useful to be able to discover the heirarchial geographic levels in a country.

In [5]:
usa.levels

geo_name                  geo_alias col_id col_name  \
0              block_groups               Block Groups     ID     NAME   
1             census_tracts              Census Tracts     ID     NAME   
2   cities_and_towns_places  Cities and Towns (Places)     ID     NAME   
3                 zip_codes                  ZIP Codes     ID     NAME   
4       county_subdivisions        County Subdivisions     ID     NAME   
5                  counties                   Counties     ID     NAME   
6                     cbsas                      CBSAs     ID     NAME   
7   congressional_districts    Congressional Districts     ID     NAME   
8                      dmas                       DMAs     ID     NAME   
9                    states                     States     ID     NAME   
10           entire_country             Entire Country     ID     NAME   

                                   feature_class_path  
0   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
1   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
2   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
3   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
4   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
5   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
6   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
7   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
8   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
9   D:\arcgis\ba_data\us_2020\Data\Demographic Dat...  
10  D:\arcgis\ba_data\us_2020\Data\Demographic Dat...

Also, access to discover the enrichment variables along with all the ways to reference these variables is extremely useful.

In [11]:
ev = usa.enrich_variables

ev

name                                             alias  \
0        AGE0_CY                            2020 Population Age <1   
1        AGE1_CY                             2020 Population Age 1   
2        AGE2_CY                             2020 Population Age 2   
3        AGE3_CY                             2020 Population Age 3   
4        AGE4_CY                             2020 Population Age 4   
...          ...                                               ...   
9222  ACSRMV2000          2018 RHHs/Moved In: 2000-2009 (ACS 5-Yr)   
9223  ACSRMV1990          2018 RHHs/Moved In: 1990-1999 (ACS 5-Yr)   
9224  ACSRMV1989        2018 RHHs/Moved In: 1989/Before (ACS 5-Yr)   
9225  ACSMEDYRMV  2018 Median Year Householder Moved In (ACS 5-Yr)   
9226    ACSOWNER                  2018 Owner Households (ACS 5-Yr)   

      data_collection              enrich_name         enrich_field_name  
0     1yearincrements  1yearincrements.AGE0_CY  F1yearincrements_AGE0_CY  
1     1yearincrements  1yearincrements.AGE1_CY  F1yearincrements_AGE1_CY  
2     1yearincrements  1yearincrements.AGE2_CY  F1yearincrements_AGE2_CY  
3     1yearincrements  1yearincrements.AGE3_CY  F1yearincrements_AGE3_CY  
4     1yearincrements  1yearincrements.AGE4_CY  F1yearincrements_AGE4_CY  
...               ...                      ...                       ...  
9222      yearmovedin   yearmovedin.ACSRMV2000    yearmovedin_ACSRMV2000  
9223      yearmovedin   yearmovedin.ACSRMV1990    yearmovedin_ACSRMV1990  
9224      yearmovedin   yearmovedin.ACSRMV1989    yearmovedin_ACSRMV1989  
9225      yearmovedin   yearmovedin.ACSMEDYRMV    yearmovedin_ACSMEDYRMV  
9226      yearmovedin     yearmovedin.ACSOWNER      yearmovedin_ACSOWNER  

[9227 rows x 5 columns]

For instance, if we are only interested in key variables for the current observed year (2019), we can reduce our subset to just these variables. Typically, this is a very useful way to get started and see what type of results a preliminary exploratory modeling effort can discover.

In [12]:
ev[
    (ev.data_collection.str.lower().str.contains('key'))  # get the key variables
    & (ev.name.str.endswith('CY'))                     # just current year (2019) variables
].reset_index(drop=True)

name                              alias data_collection  \
0    TOTPOP_CY              2020 Total Population      KeyUSFacts   
1     GQPOP_CY     2020 Group Quarters Population      KeyUSFacts   
2   DIVINDX_CY               2020 Diversity Index      KeyUSFacts   
3     TOTHH_CY              2020 Total Households      KeyUSFacts   
4   AVGHHSZ_CY        2020 Average Household Size      KeyUSFacts   
5   MEDHINC_CY       2020 Median Household Income      KeyUSFacts   
6   AVGHINC_CY      2020 Average Household Income      KeyUSFacts   
7       PCI_CY             2020 Per Capita Income      KeyUSFacts   
8     TOTHU_CY           2020 Total Housing Units      KeyUSFacts   
9     OWNER_CY            2020 Owner Occupied HUs      KeyUSFacts   
10   RENTER_CY           2020 Renter Occupied HUs      KeyUSFacts   
11   VACANT_CY          2020 Vacant Housing Units      KeyUSFacts   
12   MEDVAL_CY             2020 Median Home Value      KeyUSFacts   
13   AVGVAL_CY            2020 Average Home Value      KeyUSFacts   
14  POPGRW10CY  2010-2020 Growth Rate: Population      KeyUSFacts   
15   HHGRW10CY  2010-2020 Growth Rate: Households      KeyUSFacts   
16  FAMGRW10CY    2010-2020 Growth Rate: Families      KeyUSFacts   
17     DPOP_CY      2020 Total Daytime Population      KeyUSFacts   
18  DPOPWRK_CY          2020 Daytime Pop: Workers      KeyUSFacts   
19  DPOPRES_CY        2020 Daytime Pop: Residents      KeyUSFacts   

              enrich_name      enrich_field_name  
0    KeyUSFacts.TOTPOP_CY   KeyUSFacts_TOTPOP_CY  
1     KeyUSFacts.GQPOP_CY    KeyUSFacts_GQPOP_CY  
2   KeyUSFacts.DIVINDX_CY  KeyUSFacts_DIVINDX_CY  
3     KeyUSFacts.TOTHH_CY    KeyUSFacts_TOTHH_CY  
4   KeyUSFacts.AVGHHSZ_CY  KeyUSFacts_AVGHHSZ_CY  
5   KeyUSFacts.MEDHINC_CY  KeyUSFacts_MEDHINC_CY  
6   KeyUSFacts.AVGHINC_CY  KeyUSFacts_AVGHINC_CY  
7       KeyUSFacts.PCI_CY      KeyUSFacts_PCI_CY  
8     KeyUSFacts.TOTHU_CY    KeyUSFacts_TOTHU_CY  
9     KeyUSFacts.OWNER_CY    KeyUSFacts_OWNER_CY  
10   KeyUSFacts.RENTER_CY   KeyUSFacts_RENTER_CY  
11   KeyUSFacts.VACANT_CY   KeyUSFacts_VACANT_CY  
12   KeyUSFacts.MEDVAL_CY   KeyUSFacts_MEDVAL_CY  
13   KeyUSFacts.AVGVAL_CY   KeyUSFacts_AVGVAL_CY  
14  KeyUSFacts.POPGRW10CY  KeyUSFacts_POPGRW10CY  
15   KeyUSFacts.HHGRW10CY   KeyUSFacts_HHGRW10CY  
16  KeyUSFacts.FAMGRW10CY  KeyUSFacts_FAMGRW10CY  
17     KeyUSFacts.DPOP_CY     KeyUSFacts_DPOP_CY  
18  KeyUSFacts.DPOPWRK_CY  KeyUSFacts_DPOPWRK_CY  
19  KeyUSFacts.DPOPRES_CY  KeyUSFacts_DPOPRES_CY

## Using ArcGIS Online

While it is common to work with a primary country locally, for quite a few large corporations, once they have a successful model running for their primary country, typically the United States, then they want to start experimenting with their international markets. The `modeling` module provides a single interface to explore these hypotheticals without having to install the data locally or learn a new API or interface.

Using the `GIS` object instance connected to ArcGIS Online, we can investigate what countries are available.

In [8]:
agol = GIS(
    os.getenv('ESRI_GIS_URL'),
    username=os.getenv('ESRI_GIS_USERNAME'),
    password=os.getenv('ESRI_GIS_PASSWORD')
)

cntry_df = get_countries(agol)

cntry_df

iso2 iso3 country_name    country_id                           alt_name  \
0     AL  ALB      Albania  ALB_MBR_2019                            ALBANIA   
1     DZ  DZA      Algeria  DZA_MBR_2019                            ALGERIA   
2     AD  AND      Andorra  AND_MBR_2019                            ANDORRA   
3     AO  AGO       Angola  AGO_MBR_2019                             ANGOLA   
4     AR  ARG    Argentina  ARG_MBR_2020                          ARGENTINA   
..   ...  ...          ...           ...                                ...   
131   UY  URY      Uruguay  URY_MBR_2020                            URUGUAY   
132   UZ  UZB   Uzbekistan  UZB_MBR_2020                         UZBEKISTAN   
133   VE  VEN    Venezuela  VEN_MBR_2020  VENEZUELA, BOLIVARIAN REPUBLIC OF   
134   VN  VNM      Vietnam  VNM_MBR_2020                           VIET NAM   
135   ZM  ZMB       Zambia  ZMB_MBR_2019                             ZAMBIA   

         continent  
0           Europe  
1           Africa  
2           Europe  
3           Africa  
4    South America  
..             ...  
131  South America  
132           Asia  
133  South America  
134           Asia  
135         Africa  

[136 rows x 6 columns]

With 136 countires available, rather than scanning the entire dataframe, we can simply see if `GBR` the ISO3 code for Great Britan is available.

In [9]:
cntry_df.iso3.str.contains('GBR').any()

True

In [10]:
cntry_df[cntry_df.iso3.str.contains('GBR')]

iso2 iso3    country_name    country_id        alt_name continent
129   GB  GBR  United Kingdom  GBR_MBR_2019  UNITED KINGDOM    Europe

Since it is, we can create a `Country` object instance just like before, and find the key facts for Great Britian. Obviously each country is going to have a different set of key facts based on what is available. Also, there are a few more columns displaying simply because there are a few more columns available from the REST endpoint than are easily available through local introspection. This, though, is something we are working on to expose locally as well.

In [11]:
gbr = Country('GBR', agol)

gbr_ev = gbr.enrich_variables

gbr_ev[
    (gbr_ev.data_collection.str.lower().str.contains('keyfacts'))
    & (gbr_ev.name.str.endswith('CY'))
].reset_index(drop=True)

name                                              alias  \
0    TOTPOP_CY                              2019 Total Population   
1   POPDENS_CY               2019 Population Density (per sq. km)   
2    POPPRM_CY                           2019 Population Per Mill   
3     MALES_CY                         2019 Total Male Population   
4   FEMALES_CY                       2019 Total Female Population   
5     TOTHH_CY                              2019 Total Households   
6   AVGHHSZ_CY                        2019 Average Household Size   
7    PAGE01_CY                     2019 Total Population Age 0-14   
8    PAGE02_CY                    2019 Total Population Age 15-29   
9    PAGE03_CY                    2019 Total Population Age 30-44   
10   PAGE04_CY                    2019 Total Population Age 45-59   
11   PAGE05_CY                      2019 Total Population Age 60+   
12   MAGE01_CY                      2019 Male Population Age 0-14   
13   MAGE02_CY                     2019 Male Population Age 15-29   
14   MAGE03_CY                     2019 Male Population Age 30-44   
15   MAGE04_CY                     2019 Male Population Age 45-59   
16   MAGE05_CY                       2019 Male Population Age 60+   
17   FAGE01_CY                    2019 Female Population Age 0-14   
18   FAGE02_CY                   2019 Female Population Age 15-29   
19   FAGE03_CY                   2019 Female Population Age 30-44   
20   FAGE04_CY                   2019 Female Population Age 45-59   
21   FAGE05_CY                     2019 Female Population Age 60+   
22   HINC01_CY             2019 HHs: 1st Quintile (Below £20,808)   
23   HINC02_CY        2019 HHs: 2nd Quintile (£20,808 to £30,641)   
24   HINC03_CY        2019 HHs: 3rd Quintile (£30,642 to £41,363)   
25   HINC04_CY        2019 HHs: 4th Quintile (£41,364 to £59,350)   
26   HINC05_CY         2019 HHs: 5th Quintile (£59,351 and above)   
27  HTYP01A_CY                            2019 HHs: Single Person   
28  HTYP02A_CY   2019 HHs: Married Couple with Dependent Children   
29  HTYP03A_CY          2019 HHs: Married Couple with No Children   
30  HTYP04A_CY  2019 HHs: Cohabiting Couple w/ Dependent Children   
31  HTYP05A_CY       2019 HHs: Cohabiting Couple with No Children   
32  HTYP06A_CY    2019 HHs: Single Parent with Dependent Children   
33  HTYP07A_CY             2019 HHs: Single Parent w/ No Children   
34  HTYP08A_CY                   2019 HHs: Multi-Person Household   
35   MRST01_CY                        2019 Marital Status: Single   
36   MRST02_CY                       2019 Marital Status: Married   
37   MRST03_CY                      2019 Marital Status: Divorced   
38   MRST04_CY                       2019 Marital Status: Widowed   
39   EDUC01_CY                2019 Pop 16+/Edu: No Qualifications   
40   EDUC02_CY           2019 Pop 16+/Edu: Level 1 Qualifications   
41   EDUC03_CY           2019 Pop 16+/Edu: Level 2 Qualifications   
42   EDUC04_CY           2019 Pop 16+/Edu: Level 3 Qualifications   
43   EDUC05_CY   2019 Pop 16+/Edu: Level 4 Qualifications & Above   
44   EDUC06_CY               2019 Pop 16+/Edu: Full Time Students   
45    UNEMP_CY                         2018 Unemployed Population   
46       PP_CY                       2019 Purchasing Power: Total   
47    PPPRM_CY                    2019 Purchasing Power: Per Mill   
48     PPPC_CY                  2019 Purchasing Power: Per Capita   
49    PPIDX_CY                       2019 Purchasing Power: Index   

   data_collection                                        description vintage  \
0         KeyFacts                              2019 Total Population    2019   
1         KeyFacts  2019 Population Density (Population per Square...    2019   
2         KeyFacts                           2019 Population Per Mill    2019   
3         KeyFacts                         2019 Total Male Population    2019   
4         KeyFacts                       2019 Total Female Population    2019   
5   

## ArcGIS Enterprise (Busines Analyst Server)

Within an organization, if more than just a few data scientists are working with the data, it begins to make sense to migrate local analysis workflows to reference Business Analyst Server as part of an ArcGIS Enterprise installation instead of using ArcGIS Pro with Business Analyst or ArcGIS Online. ArcGIS Pro with Busines Analyst simply does not scale out very well technically or fiscally. ArcGIS Enterprise simply does not scale fiscally. Every time enrichment is run, it costs credits. Thus, it makes sense to quickly be able to move to Business Analyst Server.

In [12]:
prtl = GIS(
    os.getenv('ESRI_PORTAL_URL'),
    username=os.getenv('ESRI_PORTAL_USERNAME'),
    password=os.getenv('ESRI_PORTAL_PASSWORD')
)

get_countries(prtl)

iso2 iso3   country_name     country_id       alt_name      continent
0   US  USA  United States  USA_ESRI_2020  UNITED STATES  North America

In [13]:
usa = Country('USA', prtl)

usa

<modeling.Country - USA (GIS at https://geoai-ent.bd.esri.com/portal/ logged in as jmccune)>

In [14]:
usa_ev = usa.enrich_variables

usa_ev

name                                              alias  \
0      AGE0_CY                             2020 Population Age <1   
1      AGE1_CY                              2020 Population Age 1   
2      AGE2_CY                              2020 Population Age 2   
3      AGE3_CY                              2020 Population Age 3   
4      AGE4_CY                              2020 Population Age 4   
..         ...                                                ...   
37  MOEMEDYRMV  2018 Median Year Householder Moved In MOE (ACS...   
38  RELMEDYRMV  2018 Median Year Householder Moved In REL (ACS...   
39    ACSOWNER                   2018 Owner Households (ACS 5-Yr)   
40    MOEOWNER               2018 Owner Households MOE (ACS 5-Yr)   
41    RELOWNER               2018 Owner Households REL (ACS 5-Yr)   

    data_collection                                        description  \
0   1yearincrements                2020 Total Population Age <1 (Esri)   
1   1yearincrements                 2020 Total Population Age 1 (Esri)   
2   1yearincrements                 2020 Total Population Age 2 (Esri)   
3   1yearincrements                 2020 Total Population Age 3 (Esri)   
4   1yearincrements                 2020 Total Population Age 4 (Esri)   
..              ...                                                ...   
37      yearmovedin  2018 Median Year Householder Moved into Unit M...   
38      yearmovedin  2018 Median Year Householder Moved into Unit R...   
39      yearmovedin                   2018 Owner Households (ACS 5-Yr)   
40      yearmovedin               2018 Owner Households MOE (ACS 5-Yr)   
41      yearmovedin               2018 Owner Households REL (ACS 5-Yr)   

      vintage  units  
0        2020  count  
1        2020  count  
2        2020  count  
3        2020  count  
4        2020  count  
..        ...    ...  
37  2014-2018  count  
38  2014-2018  count  
39  2014-2018  count  
40  2014-2018  count  
41  2014-2018  count  

[17474 rows x 6 columns]

In [15]:
usa_ev[
    (usa_ev.data_collection.str.lower().str.contains('key'))
    & (usa_ev.name.str.endswith('CY'))
].reset_index(drop=True)

name                              alias data_collection  \
0    TOTPOP_CY              2020 Total Population      KeyUSFacts   
1     GQPOP_CY     2020 Group Quarters Population      KeyUSFacts   
2   DIVINDX_CY               2020 Diversity Index      KeyUSFacts   
3     TOTHH_CY              2020 Total Households      KeyUSFacts   
4   AVGHHSZ_CY        2020 Average Household Size      KeyUSFacts   
5   MEDHINC_CY       2020 Median Household Income      KeyUSFacts   
6   AVGHINC_CY      2020 Average Household Income      KeyUSFacts   
7       PCI_CY             2020 Per Capita Income      KeyUSFacts   
8     TOTHU_CY           2020 Total Housing Units      KeyUSFacts   
9     OWNER_CY            2020 Owner Occupied HUs      KeyUSFacts   
10   RENTER_CY           2020 Renter Occupied HUs      KeyUSFacts   
11   VACANT_CY          2020 Vacant Housing Units      KeyUSFacts   
12   MEDVAL_CY             2020 Median Home Value      KeyUSFacts   
13   AVGVAL_CY            2020 Average Home Value      KeyUSFacts   
14  POPGRW10CY  2010-2020 Growth Rate: Population      KeyUSFacts   
15   HHGRW10CY  2010-2020 Growth Rate: Households      KeyUSFacts   
16  FAMGRW10CY    2010-2020 Growth Rate: Families      KeyUSFacts   
17     DPOP_CY      2020 Total Daytime Population      KeyUSFacts   
18  DPOPWRK_CY          2020 Daytime Pop: Workers      KeyUSFacts   
19  DPOPRES_CY        2020 Daytime Pop: Residents      KeyUSFacts   

                                        description vintage     units  
0                      2020 Total Population (Esri)    2020     count  
1             2020 Group Quarters Population (Esri)    2020     count  
2                       2020 Diversity Index (Esri)    2020     count  
3                      2020 Total Households (Esri)    2020     count  
4                2020 Average Household Size (Esri)    2020     count  
5               2020 Median Household Income (Esri)    2020  currency  
6              2020 Average Household Income (Esri)    2020  currency  
7                     2020 Per Capita Income (Esri)    2020  currency  
8                   2020 Total Housing Units (Esri)    2020     count  
9          2020 Owner Occupied Housing Units (Esri)    2020     count  
10        2020 Renter Occupied Housing Units (Esri)    2020     count  
11                 2020 Vacant Housing Units (Esri)    2020     count  
12                    2020 Median Home Value (Esri)    2020  currency  
13                   2020 Average Home Value (Esri)    2020  currency  
14  2010-2020 Population: Annual Growth Rate (Esri)    2020       pct  
15  2010-2020 Households: Annual Growth Rate (Esri)    2020       pct  
16    2010-2020 Families: Annual Growth Rate (Esri)    2020       pct  
17             2020 Total Daytime Population (Esri)    2020     count  
18          2020 Daytime Population: Workers (Esri)    2020     count  
19        2020 Daytime Population: Residents (Esri)    2020     count

## Flexiblity Illustrated

While each of the above examples illustrated different data sources, just to illustrate the ease of moving between these. Here is the same workflow for the United States in ArcGIS Online and with local data.

In [16]:
usa = Country('USA', agol)
usa_ev = usa.enrich_variables
usa_ev[
    (usa_ev.data_collection.str.lower().str.contains('key'))
    & (usa_ev.name.str.endswith('CY'))
].reset_index(drop=True)

name                              alias data_collection  \
0    TOTPOP_CY              2020 Total Population      KeyUSFacts   
1     GQPOP_CY     2020 Group Quarters Population      KeyUSFacts   
2   DIVINDX_CY               2020 Diversity Index      KeyUSFacts   
3     TOTHH_CY              2020 Total Households      KeyUSFacts   
4   AVGHHSZ_CY        2020 Average Household Size      KeyUSFacts   
5   MEDHINC_CY       2020 Median Household Income      KeyUSFacts   
6   AVGHINC_CY      2020 Average Household Income      KeyUSFacts   
7       PCI_CY             2020 Per Capita Income      KeyUSFacts   
8     TOTHU_CY           2020 Total Housing Units      KeyUSFacts   
9     OWNER_CY            2020 Owner Occupied HUs      KeyUSFacts   
10   RENTER_CY           2020 Renter Occupied HUs      KeyUSFacts   
11   VACANT_CY          2020 Vacant Housing Units      KeyUSFacts   
12   MEDVAL_CY             2020 Median Home Value      KeyUSFacts   
13   AVGVAL_CY            2020 Average Home Value      KeyUSFacts   
14  POPGRW10CY  2010-2020 Growth Rate: Population      KeyUSFacts   
15   HHGRW10CY  2010-2020 Growth Rate: Households      KeyUSFacts   
16  FAMGRW10CY    2010-2020 Growth Rate: Families      KeyUSFacts   
17     DPOP_CY      2020 Total Daytime Population      KeyUSFacts   
18  DPOPWRK_CY          2020 Daytime Pop: Workers      KeyUSFacts   
19  DPOPRES_CY        2020 Daytime Pop: Residents      KeyUSFacts   

                                        description vintage     units  
0                      2020 Total Population (Esri)    2020     count  
1             2020 Group Quarters Population (Esri)    2020     count  
2                       2020 Diversity Index (Esri)    2020     count  
3                      2020 Total Households (Esri)    2020     count  
4                2020 Average Household Size (Esri)    2020     count  
5               2020 Median Household Income (Esri)    2020  currency  
6              2020 Average Household Income (Esri)    2020  currency  
7                     2020 Per Capita Income (Esri)    2020  currency  
8                   2020 Total Housing Units (Esri)    2020     count  
9          2020 Owner Occupied Housing Units (Esri)    2020     count  
10        2020 Renter Occupied Housing Units (Esri)    2020     count  
11                 2020 Vacant Housing Units (Esri)    2020     count  
12                    2020 Median Home Value (Esri)    2020  currency  
13                   2020 Average Home Value (Esri)    2020  currency  
14  2010-2020 Population: Annual Growth Rate (Esri)    2020       pct  
15  2010-2020 Households: Annual Growth Rate (Esri)    2020       pct  
16    2010-2020 Families: Annual Growth Rate (Esri)    2020       pct  
17             2020 Total Daytime Population (Esri)    2020     count  
18          2020 Daytime Population: Workers (Esri)    2020     count  
19        2020 Daytime Population: Residents (Esri)    2020     count

In [17]:
usa = Country('USA', 'local')
usa_ev = usa.enrich_variables
usa_ev[
    (usa_ev.data_collection.str.lower().str.contains('key'))
    & (usa_ev.name.str.endswith('CY'))
].reset_index(drop=True)

name                              alias data_collection  \
0    TOTPOP_CY              2020 Total Population      KeyUSFacts   
1     GQPOP_CY     2020 Group Quarters Population      KeyUSFacts   
2   DIVINDX_CY               2020 Diversity Index      KeyUSFacts   
3     TOTHH_CY              2020 Total Households      KeyUSFacts   
4   AVGHHSZ_CY        2020 Average Household Size      KeyUSFacts   
5   MEDHINC_CY       2020 Median Household Income      KeyUSFacts   
6   AVGHINC_CY      2020 Average Household Income      KeyUSFacts   
7       PCI_CY             2020 Per Capita Income      KeyUSFacts   
8     TOTHU_CY           2020 Total Housing Units      KeyUSFacts   
9     OWNER_CY            2020 Owner Occupied HUs      KeyUSFacts   
10   RENTER_CY           2020 Renter Occupied HUs      KeyUSFacts   
11   VACANT_CY          2020 Vacant Housing Units      KeyUSFacts   
12   MEDVAL_CY             2020 Median Home Value      KeyUSFacts   
13   AVGVAL_CY            2020 Average Home Value      KeyUSFacts   
14  POPGRW10CY  2010-2020 Growth Rate: Population      KeyUSFacts   
15   HHGRW10CY  2010-2020 Growth Rate: Households      KeyUSFacts   
16  FAMGRW10CY    2010-2020 Growth Rate: Families      KeyUSFacts   
17     DPOP_CY      2020 Total Daytime Population      KeyUSFacts   
18  DPOPWRK_CY          2020 Daytime Pop: Workers      KeyUSFacts   
19  DPOPRES_CY        2020 Daytime Pop: Residents      KeyUSFacts   

              enrich_name      enrich_field_name  
0    KeyUSFacts.TOTPOP_CY   KeyUSFacts_TOTPOP_CY  
1     KeyUSFacts.GQPOP_CY    KeyUSFacts_GQPOP_CY  
2   KeyUSFacts.DIVINDX_CY  KeyUSFacts_DIVINDX_CY  
3     KeyUSFacts.TOTHH_CY    KeyUSFacts_TOTHH_CY  
4   KeyUSFacts.AVGHHSZ_CY  KeyUSFacts_AVGHHSZ_CY  
5   KeyUSFacts.MEDHINC_CY  KeyUSFacts_MEDHINC_CY  
6   KeyUSFacts.AVGHINC_CY  KeyUSFacts_AVGHINC_CY  
7       KeyUSFacts.PCI_CY      KeyUSFacts_PCI_CY  
8     KeyUSFacts.TOTHU_CY    KeyUSFacts_TOTHU_CY  
9     KeyUSFacts.OWNER_CY    KeyUSFacts_OWNER_CY  
10   KeyUSFacts.RENTER_CY   KeyUSFacts_RENTER_CY  
11   KeyUSFacts.VACANT_CY   KeyUSFacts_VACANT_CY  
12   KeyUSFacts.MEDVAL_CY   KeyUSFacts_MEDVAL_CY  
13   KeyUSFacts.AVGVAL_CY   KeyUSFacts_AVGVAL_CY  
14  KeyUSFacts.POPGRW10CY  KeyUSFacts_POPGRW10CY  
15   KeyUSFacts.HHGRW10CY   KeyUSFacts_HHGRW10CY  
16  KeyUSFacts.FAMGRW10CY  KeyUSFacts_FAMGRW10CY  
17     KeyUSFacts.DPOP_CY     KeyUSFacts_DPOP_CY  
18  KeyUSFacts.DPOPWRK_CY  KeyUSFacts_DPOPWRK_CY  
19  KeyUSFacts.DPOPRES_CY  KeyUSFacts_DPOPRES_CY